dockernet/ts-tests/test/utils.ts

```diff
@@ -0,0 +1,210 @@
+ * @returns {Coin[]} An array of Coin objects
+ */
+export const coinsFromString = (coinsAsString: string): Coin[] =>
+  coinsAsString.split(",").map(coinFromString);
+
+/**
+ * Creates a StdFee object from the given gas limit and gas price.
+ *
+ * @param {number} gasLimit The gas limit to use for the fee calculation.
+ * @param {number} [gasPrice=0.025] The gas price to use for the fee calculation. Defaults to 0.025.
+ * @returns {StdFee} A StdFee object with the calculated fee amount and gas limit.
+ */
+export const feeFromGas = (
+  gasLimit: number,
+  gasPrice: number = 0.025,
+): StdFee => ({
+  amount: coinsFromString(`${gasLimit * gasPrice}ustrd`),
+  gas: String(gasLimit),
+});
+
+/**
+ * Compute the IBC denom of a token that was sent over IBC.
+ *
+ * @example
+ * To get the IBC denom of STRD on mainnet Osmosis:
+ * ```
+ * ibcDenom([{incomingPortId: "transfer", incomingChannelId: "channel-326"}], "ustrd")
+ * ```
+ *
+ * @param {Object[]} paths An array of objects containing information about the IBC transfer paths.
+ * @param {string} coinMinimalDenom The minimal denom of the coin.
+ * @returns {string} The computed IBC denom of the token.
+ */
+export const ibcDenom = (
+  paths: {
+    incomingPortId: string;
+    incomingChannelId: string;
+  }[],
+  coinMinimalDenom: string,
+): string => {
+  const prefixes: string[] = [];
+  for (const path of paths) {
+    prefixes.push(`${path.incomingPortId}/${path.incomingChannelId}`);
+  }
+
+  const prefix = prefixes.join("/");
+  const denom = `${prefix}/${coinMinimalDenom}`;
+
+  return "ibc/" + toHex(sha256(toUtf8(denom))).toUpperCase();
+};
+
+/**
+ * Convert a secp256k1 compressed public key to an address
+ *
+ * @param {Uint8Array} pubkey The account's pubkey, should be 33 bytes (compressed secp256k1)
+ * @param {String} [prefix="stride"] The address' bech32 prefix. Defaults to `"stride"`.
+ * @returns the account's address
+ */
+export function pubkeyToAddress(
+  pubkey: Uint8Array,
+  prefix: string = "stride",
+): string {
+  return bech32.encode(prefix, bech32.toWords(ripemd160(sha256(pubkey))));
+}
+
+/**
+ * Convert a secp256k1 compressed public key to an address
+ *
+ * @param {Uint8Array} pubkey The account's pubkey as base64 string, should be 33 bytes (compressed secp256k1)
+ * @param {String} [prefix="stride"] The address' bech32 prefix. Defaults to `"stride"`.
+ * @returns the account's address
+ */
+export function base64PubkeyToAddress(
+  pubkey: string,
+  prefix: string = "stride",
+): string {
+  return pubkeyToAddress(fromBase64(pubkey), prefix);
+}
+
+/**
+ * Convert self delegator address to validator address
+ *
+ * @param {String} selfDelegator The self delegator bech32 encoded address
+ * @param {String} [prefix="stride"] The self delegator address' bech32 prefix. Defaults to `"stride"`.
+ * @returns the account's address
+ */
+export function selfDelegatorAddressToValidatorAddress(
```

Comment:
by "self delegator" does this just mean a regular non-validator address?

Reply:
that's the cosmos term for the account that controls a validator. it's basically the same bytes but with a different prefix: `stride` vs `stridevaloper`